client/lib/utils/coerce-values.js

```diff
@@ -0,0 +1,85 @@
+import coerce from 'type-coerce';
+
+/**
+ * Retrieve a field's schema, handling referenced schema definitions if need be.
+ *
+ * @param {Object} fieldSchema - Schema object for field.
+ * @param {Object} definitions - Common definitions.
+ * @returns {Object} - Schema object for field, potentially pulled from $ref's definition.
+ */
+const getFieldSchema = ( fieldSchema, definitions ) => {
+	if ( fieldSchema.hasOwnProperty( '$ref' ) ) {
+		const definitionKey = fieldSchema.$ref.match( /^#\/definitions\/(.+)/ );
+
+		if ( definitionKey && definitions.hasOwnProperty( definitionKey[ 1 ] ) ) {
+			fieldSchema = definitions[ definitionKey[ 1 ] ];
+		}
+	}
+	return fieldSchema;
+};
+
+/**
+ * Coerces a value into a type.
+ *
+ * @param {*} value - Value to coerce.
+ * @param {string} type - Type to coerce value to.
+ * @returns {*} - Coerced value.
+ */
+const coerceValue = ( value, type ) => {
+	if ( 'number' === type ) {
+		if ( '' === value ) {
+			return undefined;
+		}
+
+		if ( ! isNaN( value ) ) {
+			return parseFloat( value );
+		}
+	} else if ( coerce.hasOwnProperty( type ) ) {
+		return coerce[ type ]( value );
+	}
+	return value;
+};
+
+/**
+ * Coerce a form values object using the provided schema.
+ *
+ * @param {Object} schema - Schema containing type declarations.
+ * @param {Object} values - Form values.
+ * @param {Object} [definitions=null] - Optional. Schema definitions, parsed from schema if omitted.
```

Comment:
Is there any use case for this 3rd argument? If not, it's better to just remove it and add it later if needed.